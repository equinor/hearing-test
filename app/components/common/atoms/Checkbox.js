import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const styles = {
  defaultButtonStyle: {
    backgroundColor: 'rgba(0,0,0,0)',
  },
};

export default class Checkbox extends Component {
  static propTypes = {
    onValueChange: PropTypes.func.isRequired,
    checked: PropTypes.bool,
    disabled: PropTypes.bool,
    size: PropTypes.number,
    color: PropTypes.string,
    style: PropTypes.object,
  };

  static defaultProps = {
    checked: false,
    disabled: false,
    style: {},
    size: 24,
    color: '#0185B7',
  };

  render() {
    return (
      <TouchableOpacity
        style={[styles.defaultButtonStyle, this.props.style]}
        onPress={() => {
          this.props.onValueChange(this.props.disabled ? this.props.checked : !this.props.checked);
        }}
        disabled={this.props.disabled}
      >
        <Icon
          name={this.props.checked ? 'check-box' : 'check-box-outline-blank'}
          size={this.props.size}
          color={this.props.color}
        />
      </TouchableOpacity>
    );
  }
}
