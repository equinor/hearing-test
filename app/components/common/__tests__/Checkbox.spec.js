import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import Checkbox from '../atoms/Checkbox';

it('renders and behaves correctly', () => {
  let checked = false;
  const callback = value => {
    checked = value;
  };
  const onCheckboxPress = sinon.spy(callback);
  const wrapper = shallow(<Checkbox onValueChange={onCheckboxPress} />);
  const checkboxControl = wrapper.find('TouchableOpacity');
  expect(wrapper).toMatchSnapshot();
  expect(onCheckboxPress.calledOnce).toBeFalsy();
  checkboxControl.simulate('press');
  expect(onCheckboxPress.calledOnce).toBeTruthy();
  expect(checked).toBe(true);
  wrapper.setProps({ checked: false, disabled: true });
  checkboxControl.simulate('press');
  expect(checked).toBe(false);
});
