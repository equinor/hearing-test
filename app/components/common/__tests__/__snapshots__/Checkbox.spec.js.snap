// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and behaves correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    checked={false}
    color="#0185B7"
    disabled={false}
    onValueChange={[Function]}
    size={24}
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Icon
        allowFontScaling={false}
        color="#0185B7"
        name="check-box-outline-blank"
        size={24}
      />,
      "disabled": false,
      "onPress": [Function],
      "style": Array [
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
        },
        Object {},
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": false,
        "color": "#0185B7",
        "name": "check-box-outline-blank",
        "size": 24,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Icon
          allowFontScaling={false}
          color="#0185B7"
          name="check-box-outline-blank"
          size={24}
        />,
        "disabled": false,
        "onPress": [Function],
        "style": Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
          },
          Object {},
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "color": "#0185B7",
          "name": "check-box-outline-blank",
          "size": 24,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
